### PR TITLE
test: grouped constructor tests and added missing test

### PR DESCRIPTION
**Pull request by alestiago**
## Status

**READY**

## Breaking Changes

NO

## Description

Changes:
- Grouped constructor related test inside a group. In order to improve readability and maintainability of the code.
- Included missing test case scenario where state is not null and controller is null

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore
- [X] 🧪 Testing

### CODE REVIEW

File: test/flow_builder_test.dart [hunk 1]
```diff
@@ -5,35 +5,48 @@ import 'package:flutter_test/flutter_test.dart';
...
+        expect(
+          () => FlowBuilder(
+            onGeneratePages: (dynamic _, List<Page<dynamic>> __) => [],
+            controller: FlowController(''),
+          ),
+          isNot(throwsAssertionError),
+        );
+      });
+
+      test('does not throw when controller is null if state is present',
```
Comment (felangel) [inline test/flow_builder_test.dart:40]: Thanks! 💯
